### PR TITLE
Add student number field

### PR DESCRIPTION
**Pull request by tzejit**
Add student number field to Student model
Add student number test
Update other tests accordingly

### CODE REVIEW

File: src/main/java/seedu/address/model/student/Student.java [hunk 1]
```diff
@@ -103,21 +110,29 @@ public boolean equals(Object other) {
         Student otherStudent = (Student) other;
         return otherStudent.getName().equals(getName())
                 && otherStudent.getEmail().equals(getEmail())
-                && otherStudent.getTags().equals(getTags());
+                && otherStudent.getStudentNumber().equals((getStudentNumber()))
+                && otherStudent.getTags().equals(getTags())
+                && otherStudent.getAttendance().equals(getAttendance());
```
Comment (wanninglim) [inline src/main/java/seedu/address/model/student/Student.java:115]: Think can remove this line on attendance as well!

File: src/main/java/seedu/address/model/student/StudentNumber.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    public static boolean isValidNumber(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+
```
Comment (wanninglim) [inline src/main/java/seedu/address/model/student/StudentNumber.java:42]: There's an extra new line here hehe

File: src/main/java/seedu/address/storage/JsonAdaptedStudent.java [hunk 3]
```diff
@@ -89,9 +94,18 @@ public Student toModelType() throws IllegalValueException {
...
+        if (!StudentNumber.isValidNumber(studentNumber)) {
+            throw new IllegalValueException(StudentNumber.MESSAGE_CONSTRAINTS);
+        }
+        final StudentNumber modelStudentNumber = new StudentNumber(studentNumber);
+
         final Set<Tag> modelTags = new HashSet<>(studentTags);
         final Attendance modelAttendance = new Attendance(studentAttendance);
 
-        return new Student(modelName, modelEmail, modelTags, modelAttendance);
+        return new Student(modelName, modelEmail, modelStudentNumber , modelTags, modelAttendance);
```
Comment (wanninglim) [inline src/main/java/seedu/address/storage/JsonAdaptedStudent.java:109]: Extra space before comma hehe... small random stuffs

File: src/main/java/seedu/address/model/student/Student.java [hunk 4]
```diff
@@ -103,21 +110,29 @@ public boolean equals(Object other) {
...
         builder.append(getName())
                 .append("; Email: ")
-                .append(getEmail());
+                .append(getEmail())
+                .append("; Student Number: ")
+                .append(getStudentNumber())
+                .append("; Attendance: ")
+                .append(getAttendance())
+                .append("; Participation: ")
+                .append(getParticipation());
```
[the remark this answers is not in the document]
Comment (tzejit) [reply src/main/java/seedu/address/model/student/Student.java]: Thanks for the comments. They are fixed now.

**Pull request merged**